Api/V8/BeanManager.php

```diff
@@ -0,0 +1,79 @@
+<?php
+namespace Api\V8;
+
+use Api\V8\Exception\BeanNotFoundException;
+
+class BeanManager
+{
+    /**
+     * @var array
+     */
+    private $beanAliases;
+
+    /**
+     * @param array $beanAliases
+     */
+    public function __construct(array $beanAliases)
+    {
+        $this->beanAliases = $beanAliases;
+    }
+
+    /**
+     * @param string $module
+     * @return \SugarBean
+     * @throws \InvalidArgumentException When the module is unknown.
+     */
+    public function newBeanSafe(string $module): \SugarBean
+    {
+        $bean = \BeanFactory::newBean($module);
+        if (!$bean instanceof \SugarBean) {
+            throw new \InvalidArgumentException(sprintf('Could not get new bean for module: %s', $module));
+        }
+
+        return $bean;
+    }
+
+    /**
+     * @param string $module
+     * @param string $id
+     * @param array $params
+     * @param bool $deleted
+     * @return \SugarBean
+     * @throws \DomainException In case id/module is invalid, bean with this id is not found or has unexpected class.
+     * @throws BeanNotFoundException When the given combinations of params don't result a bean.
+     */
+    public function getBeanSafe(
+        string $module,
+        string $id,
+        array $params = [],
+        bool $deleted = true
+    ): \SugarBean {
```

Comment:
Please remove the return type. As we need to support php 5.5.9

Reply:
As I mentioned above, let's talk about a possible solution for that.